src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -183,16 +190,23 @@ def get_load_balancer(resource_pool)
     def create_network_interfaces(instance_id, location, resource_pool, network_configurator)
       network_interfaces = []
       load_balancer = get_load_balancer(resource_pool)
-      public_ip = get_public_ip(network_configurator.vip_network)
       networks = network_configurator.networks
       networks.each_with_index do |network, index|
-        security_group = get_network_security_group(resource_pool, network)
         nic_name = "#{instance_id}-#{index}"
+        public_ip = nil
+        if network.vip?
+          public_ip = get_public_ip(network_configurator.vip_network)
+        elsif resource_pool['assign_dynamic_public_ips']
```

Comment:
Azure CPI supports to create multiple network interfaces so that you can not assign the public IP to non-primary network interface card.